client/my-sites/hosting/github/github-connect-card/use-github-repos.ts

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from 'react-query';
+import wp from 'calypso/lib/wp';
+
+export const USE_GITHUB_REPOS_QUERY_KEY = 'github-repos-query-key';
+
+export interface Repo {
+	name: string;
+	full_name: string;
+}
+
+export const useGithubRepos = ( siteId: number | null ) => {
+	return useQuery(
+		[ USE_GITHUB_REPOS_QUERY_KEY, siteId ],
+		(): Repo[] =>
+			wp.req.get( {
+				path: `/sites/${ siteId }/hosting/github/repos`,
+				apiNamespace: 'wpcom/v2',
+			} ),
+		{
+			enabled: !! siteId,
+			select: ( data ) =>
+				data.map( ( repo: Repo ) => ( {
```

Comment:
I will simplify the server data to be a list of strings so you can remove this mapping code later on.

Reply:
maybe we should allow it as is in case we decided to user other fields from the returned object.